subsys/net/lib/rest_client/src/rest_client.c

```diff
@@ -120,23 +120,24 @@ static int rest_client_sckt_tls_setup(int fd, const char *const tls_hostname,
 	return 0;
 }
 
-static int rest_client_sckt_timeouts_set(int fd)
+static int rest_client_sckt_timeouts_set(int fd, int32_t timeout_ms)
 {
 	int err;
 	struct timeval timeout = { 0 };
 
-	if (CONFIG_REST_CLIENT_SCKT_SEND_TIMEOUT > 0) {
+	if (timeout_ms > 0 || CONFIG_REST_CLIENT_REQUEST_TIMEOUT > 0) {
 		/* Send TO also affects TCP connect */
-		timeout.tv_sec = CONFIG_REST_CLIENT_SCKT_SEND_TIMEOUT;
+
+		timeout.tv_sec = timeout_ms > 0 ? timeout_ms / 1000 :
+			CONFIG_REST_CLIENT_REQUEST_TIMEOUT;
 		err = setsockopt(fd, SOL_SOCKET, SO_SNDTIMEO, &timeout, sizeof(timeout));
 		if (err) {
 			LOG_ERR("Failed to set socket send timeout, error: %d", errno);
 			return err;
 		}
-	}
 
-	if (CONFIG_REST_CLIENT_SCKT_RECV_TIMEOUT > 0) {
-		timeout.tv_sec = CONFIG_REST_CLIENT_SCKT_RECV_TIMEOUT;
+		timeout.tv_sec = timeout_ms > 0 ? timeout_ms / 1000 :
+			CONFIG_REST_CLIENT_REQUEST_TIMEOUT;
```

Comment:
Could be removed, the timeout is the same as for SO_SNDTIMEO.

Reply:
Yep, and I was supposed to simplify previous if condition too due to latest decision to remove send and recv timeout configs. Will do it now.